docs/dev/how-to/send-message-l2-l1.md

```diff
@@ -302,3 +203,141 @@ L1 Batch for block :>>  77512
 Retrieving proof for batch 77512, transaction index 32 and proof id 14
 Result is :>>  true
 ```
+
+## Send a message
+
+Two transactions are required:
+
+- An L2 transaction which sends a message of arbitrary length.
+- An L1 read; implemented by a getter function on an L1 smart contract.
+
+5. Get a `Contract` object that represents the [`L1Messenger`](../../reference/architecture/system-contracts.md#l1messenger) contract.
+
+6. Transform the request into a raw bytes array.
+
+7. Use the [`sendToL1`](https://github.com/matter-labs/v2-testnet-contracts/blob/b8449bf9c819098cc8bfee0549ff5094456be51d/l2/system-contracts/interfaces/IL1Messenger.sol#L5) function from the [`IL1Messenger.sol`](https://github.com/matter-labs/v2-testnet-contracts/blob/b8449bf9c819098cc8bfee0549ff5094456be51d/l2/system-contracts/interfaces/IL1Messenger.sol#L4) interface, passing the message as a raw bytes array.
+
+Each sent message emits an [`L1MessageSent`](https://github.com/matter-labs/v2-testnet-contracts/blob/b8449bf9c819098cc8bfee0549ff5094456be51d/l2/system-contracts/interfaces/IL1Messenger.sol#L8) event.
+
+```solidity
+event L1MessageSent(address indexed _sender, bytes32 indexed _hash, bytes _message);
+
+function sendToL1(bytes memory _message) external returns (bytes32);
+```
+
+4.1 The return value from `sendToL1` is the `keccak256` hash of the message bytes.
+
+## Prove the result
+
+1. The [`proveL2MessageInclusion`](https://github.com/matter-labs/v2-testnet-contracts/blob/b8449bf9c819098cc8bfee0549ff5094456be51d/l1/contracts/zksync/facets/Mailbox.sol#L46) function returns a boolean parameter indicating whether the message was sent successfully to L1.
```

Comment:
This `1.` makes little sense to me unless there were more steps ahead

```suggestion
The [`proveL2MessageInclusion`](https://github.com/matter-labs/v2-testnet-contracts/blob/b8449bf9c819098cc8bfee0549ff5094456be51d/l1/contracts/zksync/facets/Mailbox.sol#L46) function returns a boolean parameter indicating whether the message was sent successfully to L1.
```

Reply:
Fixed in the latest commit